### PR TITLE
[W03] Add Artemis-tests-based testing utilities

### PR DESCRIPTION
**Pull request by Anatoly03**
#### General

- [x] I confirm that my tests, if any, are correct
    - [x] Optional: I pass **all** artemis tests.
    - [x] Optional: I pass **all** tests, that are added in this PR. [If any I don't pass, they are marked as disabled]
    - [x] Optional: I calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.
- [ ] Other students approved this Pull Request.

### Description

Artemis-Tests split into different test cases for the sake of verification of what works and what not.

tl;dr of the code

```java
public class UnitTest {
    //
    // <region help>
    //
    
    Arr, Stck, Queue, StkQu, Ints - trivial;
    
    private static Interval Ival(...);

    // Dynamic Array Testing Utility
    private static class TDA extends DynamicArray;

    // Test Utility
    private static class TestComponent<Data extends Collection>;

    // Dynamic Stack Testing Utility
    private static class TDS extends TestComponent<DynamicStack>;

    // Dynamic RingQueue Utility
    private static class TDQ extends TestComponent<RingQueue>;

    // Dynamic Stack Queue Utility
    private static class TDSQ extends StackyQueue;
    
    //
    // </region help>
    //

    //
    // <region test driven tests />
    //
    
    //
    // <region beauty tests>
    //

    more tests can be added here
    
    //
    // </region beauty tests>
    //
}
```

### CODE REVIEW

**Comment by ITegs** [comment]
Your pr title does not address the correct task (`[W03]` -> `[W04]`)

**Comment by Anatoly03** [comment]
@ITegs fixed

File: w04/test/gad/simplehash/UnitTest.java [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+
+// import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+//
+// HALT STOPP! LESEN!
+//
+// TESTS SIND GE'SPEEDRUN'T UND OHNE QUALITÄT VERFASST
+// DISABLED = ICH [=VERFASSER] PASSE ARTEMIS TESTS [=VIELLEICHT] ABER TESTS NICHT [=SICHER]
```
Comment (ITegs) [inline w04/test/gad/simplehash/UnitTest.java]: You selected
> - [x] I confirm that my tests, if any, are correct
so this sentence should be obsolete (?)

Comment (Anatoly03) [reply w04/test/gad/simplehash/UnitTest.java]: Dead code indeed

**Review by ITegs — APPROVED** [review]
LGTM
